load.php

```diff
@@ -477,17 +477,37 @@ function perflab_run_module_activation_deactivation( $old_value, $value ) {
 				return;
 			}
 
-			$current_user = get_current_user_id();
-			$dismissed    = array_filter( explode( ',', (string) get_user_meta( $current_user, 'dismissed_wp_pointers', true ) ) );
+			$current_user = wp_get_current_user();
 
-			if ( ! in_array( 'perflab-module-migration-pointer', $dismissed, true ) ) {
-				return;
-			}
+			/*
+			 * Disable WordPress pointers for specific users based on conditions.
+			 *
+			 * Checks if there is a large user count on the site. If true,
+			 * disables pointers for the current user only. Otherwise, disables
+			 * pointers for users with the same role as the current user.
+			 */
+			if ( ! wp_is_large_user_count() ) {
+				perflab_dismissed_wp_pointers( $current_user );
+			} else {
+				$current_user_roles = array_shift( $current_user->roles );
```

Comment:
The `array_shift()` is problematic as it modifies the actual `$current_user` object, which we need to avoid. On another note, the variable should probably be renamed as it only contains a single role, not multiple. Here's an alternative solution that doesn't modify the original object property.
```suggestion
				$current_user_roles = $current_user->roles;
				$current_user_role  = array_shift( $current_user_roles );
```